reconcile/utils/terraform_resource_spec.py

```diff
@@ -0,0 +1,106 @@
+from dataclasses import field
+from pydantic.dataclasses import dataclass
+import json
+from typing import Any, Mapping, Optional
+import base64
+from reconcile.utils.openshift_resource import OpenshiftResource
+
+
+@dataclass
+class TerraformResourceSpec:
+
+    resource: Mapping[str, Any]
+    namespace: Mapping[str, Any]
+    secret: Mapping[str, str] = field(init=False, default_factory=lambda: {})
+
+    @property
+    def provider(self):
+        return self.resource.get("provider")
+
+    @property
+    def identifier(self):
+        return self.resource.get("identifier")
+
+    @property
+    def account(self):
+        return self.resource.get("account")
+
+    @property
+    def namespace_name(self) -> str:
+        return self.namespace["name"]
+
+    @property
+    def cluster_name(self) -> str:
+        return self.namespace["cluster"]["name"]
+
+    @property
+    def output_prefix(self):
+        return f"{self.identifier}-{self.provider}"
+
+    @property
+    def output_resource_name(self):
+        return self.resource.get("output_resource_name") or self.output_prefix
+
+    def _annotations(self) -> dict[str, str]:
+        annotation_str = self.resource.get("annotations")
+        if annotation_str:
+            return json.loads(annotation_str)
+        else:
+            return {}
+
+    def get_secret_field(self, field: str) -> Optional[str]:
+        return self.secret.get(field)
+
+    def id_object(self) -> "TerraformResourceIdentifier":
+        return TerraformResourceIdentifier.from_dict(self.resource)
+
+    def build_oc_secret(
```

Comment:
What do you think of moving this to OpenshiftRecource like:
```
def createFromTerraformResourcespec(cls, tr: TerraformResourceSpec):
   or = cls()
   ....
```

 Cause it holds plenty of details regarding Openshift resources.